[configure] eagerly determine the truthfulness of environment variables

darthsuogles opened the pull request:
Eagerly determine the truthfulness of environment variables in `get_var` function.

This way we can skip checking, e.g. Android workspace setup if the user sets `TF_SET_ANDROID_WORKSPACE=0`

Tested manually.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

darthsuogles commented:
I signed it!

darthsuogles commented:
I signed it!

darthsuogles committed:
determine environment variable truthfulness

Signed-off-by: Philip Yang <darthsuogles@gmail.com>

caisq commented on configure.py:
```diff
@@ -319,6 +319,15 @@ def get_var(environ_cp,
   var = environ_cp.get(var_name)
+  if var is not None:
+    _var_content = var.strip().lower()
+    if _var_content in ['1', 'y', 'true']:
```
Do you want to also include 'yes' and 'no' as acceptable values?

darthsuogles replied:
Indeed. Added "yes" and "no" to acceptable values. Thanks!

caisq commented on configure.py:
```diff
@@ -319,6 +319,15 @@ def get_var(environ_cp,
+    elif _var_content in ['0', 'n', 'false']:
+      var = False
+    else:
+      var = None
```
In this case, should you error out instead?

darthsuogles replied:
Made changes to throw a `UserInputError` in this case. Thanks!

darthsuogles committed:
address PR comments

Signed-off-by: Philip Yang <darthsuogles@gmail.com>

caisq commented on configure.py:
```diff
@@ -319,6 +325,24 @@ def get_var(environ_cp,
   var = environ_cp.get(var_name)
+  if var is not None:
+    _var_content = var.strip().lower()
+    _true_contents = ['1', 'y', 'yes', 'true']
```
You can use tuples instead of lists, to avoid inadvertent changes the them.

darthsuogles replied:
Done. Thanks

darthsuogles committed:
PR comments: using tuple

Signed-off-by: Philip Yang <darthsuogles@gmail.com>

caisq commented on configure.py:
```diff
@@ -302,6 +302,12 @@ def get_var(environ_cp,
+
+  Raises:
+    UserInputError: if an environment variable is set, but it cannot be
+    interpreted as a boolean indicator, assume that the user has made a
```
Hanging indent should be 2 chars (lines 308-310)

caisq commented on configure.py:
```diff
@@ -319,6 +325,24 @@ def get_var(environ_cp,
 
   var = environ_cp.get(var_name)
+  if var is not None:
+    _var_content = var.strip().lower()
```
Google Python style doesn't require an underscore in the beginning of a local variable name. See:
https://google.github.io/styleguide/pyguide.html?showone=Naming#Naming

(although it requires an underscore at the beginning of a private function name).

caisq commented on configure.py:
```diff
@@ -319,6 +325,24 @@ def get_var(environ_cp,
+    elif _var_content in _false_contents:
+      var = False
+    else:
+      raise UserInputError('environment variable %s must be set as a boolean indicator\n'
```
Upper case for first letters of a sentence. Also end sentences with periods.

darthsuogles committed:
Address PR comments

1. Indentation on "raises"
2. Removing local variables' leading underscores.
3. Capitalize and full stop.

Signed-off-by: Philip Yang <darthsuogles@gmail.com>

caisq committed:
String constant simplification

caisq commented:
@tensorflow-jenkins test this please
